projects/plugins/protect/src/js/hooks/use-protect-data/index.js

```diff
@@ -8,6 +8,8 @@ import { useSelect } from '@wordpress/data';
  */
 import { STORE_ID } from '../../state/store';
 
+let hasUncheckedItems = false;
```

Comment:
Since it's not inserted in react lifecycle, it won't trigger updates when the value changes, is that expected?

Reply:
The only situation this would change is from disconnected to connected, in which case the response will most likely be `empty`... testing in real life this does not seem to be a problem.

I added a note to our [task](1202064326332560-as-1202183900595787) to refactor the data handling and make all the transformations in the server side. When we do that, we can also update this value in the response from the server and we won't have a problem.